spec/introduction.md

```diff
@@ -0,0 +1,430 @@
+
+```fsharp
+    let rec fib x = if x < 2 then 1 else fib(x-1) + fib(x-2)
+
+    let fibs =
+        Async.Parallel [ for i in 0..40 -> async { return fib(i) } ]
+        |> Async.RunSynchronously
+
+    printfn "The Fibonacci numbers are %A" fibs
+
+    System.Console.ReadKey(true)
+```
+
+The preceding code sample shows multiple, parallel, CPU-bound computations.
+
+F# is also a reactive language. The following example requests multiple web pages in parallel, reacts to the responses for each request, and finally returns the collected results.
+
+```fsharp
+    open System
+    open System.IO
+    open System.Net
+
+    let http url =
+        async {
+            let req = WebRequest.Create(Uri url)
+            use! resp = req.AsyncGetResponse()
+            use stream = resp.GetResponseStream()
+            use reader = new StreamReader(stream)
+            let contents = reader.ReadToEnd()
+            return contents
+            }
+
+    let sites = [
+        "http://www.bing.com"
+        "http://www.google.com"
+        "http://www.yahoo.com"
+        "http://www.search.com"
+        ]
+
+    let htmlOfSites =
+        Async.Parallel [for site in sites -> http site ]
+        |> Async.RunSynchronously
+```
+
+By using asynchronous workflows together with other CLI libraries, F# programs can implement parallel tasks, parallel I/O operations, and message-receiving agents.
+
+### Strong Typing for Floating-Point Code
+
+F# applies type checking and type inference to floating-point-intensive domains through units of measure inference and checking. This feature allows you to type-check programs that manipulate floating-point numbers that represent physical and abstract quantities in a stronger way than other typed languages, without losing any performance in your compiled code. You can think of this feature as providing a type system for floating-point code.
+
+Consider the following example:
+
+```fsharp
+    [<Measure>] type kg
+    [<Measure>] type m
+    [<Measure>] type s
+
+    let gravityOnEarth = 9.81<m/s^2>
+    let heightOfTowerOfPisa = 55.86<m>
+    let speedOfImpact = sqrt(2.0 * gravityOnEarth * heightOfTowerOfPisa)
+```
+
+The `Measure` attribute tells F# that `kg`, `s`, and `m` are not really types in the usual sense of the word, but are used to build units of measure. Here `speedOfImpact` is inferred to have type `float<m/s>`.
+
+### Object-Oriented Programming and Code Organization
+
+The sample program shown at the start of this chapter is a _script_. Although scripts are excellent for rapid prototyping, they are not suitable for larger software components. F# supports the transition from scripting to structured code through several techniques.
```

Comment:
We can use single style for markdown in such a large body of work (VSCode can do this), and we can make this a lint-check in PRs as well perhaps? Not sure how precise we should be here.
```suggestion
The sample program shown at the start of this chapter is a *script*. Although scripts are excellent for rapid prototyping, they are not suitable for larger software components. F# supports the transition from scripting to structured code through several techniques.
```

Reply:
I agree, we should use a single style